### DISCUSSION TITLE
Language Options

### DISCUSSION THREAD

**Discussion by hbshvn** [DaxStudio/DaxStudio#571] [Q&A]
Hello Community,

my Windows as well as Power BI regional settings are set to English but I still have german tooltip above functions as well as german arguments prefilled in DAX expressions.

How can I change the language to show it all in english?

**Comment by dgosbell** [comment]
The Microsoft client libraries will detect the windows locale settings and use those settings to determine what language to display for the functions. If you open the advanced options and change the locale setting to "English (United States)" this will override the language settings for the current session in DAX Studio, but the default will be picked up from the Windows settings on the PC. 

What **Country or Region** do you have configured in Windows? 

And if you've changed this recently - have you logged out and back in since changing this?

**Reply by hbshvn** [reply]
Hey Darren,

thank you for your response.
I changed my Windows Country and Region settings to UK yesterday, but still have all german arguments and tooltips.

It's also not working when I change Current File Regional Settings to English (Europe) in Power BI and then start DAX Studio.

**Reply by dgosbell** [reply]
> It's also not working when I change Current File Regional Settings to English (Europe) in Power BI and then start DAX Studio.

Yes, I would not expect that to have much impact on DAX Studio, the current file settings only affect data import operations.

It appears that what is driving this is the model language under the global settings in Power BI Desktop. Unfortunately the model language gets set when the model is first created. I just set my Power BI model language to German, then restarted Power BI and created a new test model and for that model all the error messages and function help are in German. But if I connect back to a different model created with my default English setting all the text is now in English again.

If you connect to your model and run the following query it will show you what culture your model is set to

```
select [Culture] from $SYSTEM.TMSCHEMA_MODEL
```

Unfortunately I don't believe there any way to alter the culture for an existing model, you would have to create a new model and copy all your queries and measures over to it. I just checked with Tabular editor it has this property greyed out saying that it cannot be altered once any child objects have been created.

**Reply by hbshvn** [reply]
Hello Darren,
unfortunately, it doesn't work for me.
Even if I choose US model, I still have german tooltips and args:
![image](https://user-images.githubusercontent.com/69720211/111466988-3b655080-8724-11eb-8acc-ba17cccf422f.png)

**Reply by dgosbell** [reply]
So this is all controlled by Microsoft code. In my testing it seemed that the culture of the model was driving the language, but the other thing that could be influencing is the locale of the connection (which is picked up from the client machine)

if you run the following after connecting to your model it will show what locale id your connection is using

```
select [PropertyName], [Value]
from $SYSTEM.DISCOVER_PROPERTIES
WHERE [PropertyName] = 'LocaleIdentifier'
```

I see 3081 (which is en-AU) but for en-UK is 2057 and en-US is 1033 

You can see a full list of all the locales available in windows from the connection dialog and you can override the default locale using this option if the query above does not return the locale you are expecting

![image](https://user-images.githubusercontent.com/8241830/111553237-0fa3a280-87d8-11eb-97b1-d162c2311b51.png)

If you go to the location where DAX Studio is installed there is a program called DaxStudio.Checker.exe if you run that what does it show in the Operating System section as being the culture for your machine?

**Reply by hbshvn** [reply]
Hi Darren,

Surprisingly I've got new machine this week and everything is fine there. 
I got english tooltips and arguments by default.
Thank you for support and also for great App.

**Reply by dgosbell** [reply]
Interesting, so there was probably some regional setting on the old machine somewhere. I'm glad you got this working.
